File: fn_flux.cpp
#include "fn_flux.h"

void fn_flux(Matrix<double> &phi, Matrix<double> &f)
{
	struct test_cases* pt_test = get_tests();

	int N = phi.rows();	
	Matrix<double> &delta = *pt_test->delta;
	
	double vrho = sed_hsf(sum1D(phi));
	double p2 = dot_product(phi, delta);

	for (int j = 1; j <= N; j++) f(j) = phi(j)*vrho*(delta(j) - p2);

	delete pt_test->delta;
	delete pt_test;
}

File: fn_flux.m
function f = fn_flux(phi, f)

%int: j, N
%double: rho, vrho, p2
%vector: phi, f, delta

global delta;

N = length(phi);

rho = 0;
p2 = 0;
for j = 1:N
    rho = rho + phi(j);
    p2 = p2 + phi(j)*delta(j);
end

vrho = sed_hsf(rho);

for j = 1:N
    f(j) = phi(j)*vrho*(delta(j) - p2);
end